src/pynxtools/nomad/schema.py

```diff
@@ -1074,368 +1332,103 @@ def create_package_from_nxdl_directories() -> Package:
-        setattr(python_module, section.name, section.section_cls)
-
-
-init_nexus_metainfo()
-
-
-def normalize_fabrication(self, archive, logger):
-    """Normalizer for fabrication section."""
-    current_cls = section_definitions[_rename_nx_for_nomad("NXfabrication")].section_cls
-    self.name = (
-        self.__dict__["nx_name"]
-        + " ("
-        + ((self.vendor__field + " / ") if self.vendor__field else "")
-        + (self.model__field if self.model__field else "")
-        + ")"
-    )
-    super(current_cls, self).normalize(archive, logger)
-
-
-def normalize_sample_component(self, archive, logger):
-    """Normalizer for sample_component section."""
-    current_cls = section_definitions[
-        _rename_nx_for_nomad("NXsample_component")
-    ].section_cls
-    if self.name__field:
-        self.name = self.name__field
-    else:
-        self.name = self.__dict__["nx_name"]
-    if self.mass__field:
-        self.mass = self.mass__field
-    # we may want to add normalisation for mass_fraction (calculating from components)
-    super(current_cls, self).normalize(archive, logger)
+            package.section_definitions.append(section)
 
+    return package
 
-def normalize_sample(self, archive, logger):
-    """Normalizer for sample section."""
-    current_cls = section_definitions[_rename_nx_for_nomad("NXsample")].section_cls
-    self.name = self.__dict__["nx_name"] + (
-        " (" + self.name__field + ")" if self.name__field else ""
-    )
-    # one could also copy local ids to identifier for search purposes
-    super(current_cls, self).normalize(archive, logger)
 
+nexus_metainfo_package: Optional[Package] = None  # pylint: disable=C0103
 
-def normalize_entry(self, archive, logger):
-    """Normalizer for Entry section."""
-    current_cls = section_definitions[_rename_nx_for_nomad("NXentry")].section_cls
-    if self.start_time__field:
-        self.start_time = self.start_time__field
-    self.name = self.__dict__["nx_name"] + (
-        " (" + self.title__field + ")" if self.title__field is not None else ""
-    )
-    # one could also copy local ids to identifier for search purposes
-    super(current_cls, self).normalize(archive, logger)
 
+def save_nexus_schema():
+    # global nexus_metainfo_package  # pylint: disable=global-statement
+    schema_dict = nexus_metainfo_package.m_to_dict()
 
-def normalize_process(self, archive, logger):
-    """Normalizer for Process section."""
-    current_cls = section_definitions[_rename_nx_for_nomad("NXprocess")].section_cls
-    if self.date__field:
-        self.start_time = self.date__field
-    self.name = self.__dict__["nx_name"]
-    # one could also copy local ids to identifier for search purposes
-    super(current_cls, self).normalize(archive, logger)
+    local_dir = os.path.abspath(os.path.dirname(__file__))
+    nxs_filepath = os.path.join(local_dir, "nxs_metainfo_package.json")
```

Comment:
Depending on where and who installed this package, and who executes this script, the write permission to this location may not always be possible.
Maybe choose some standard location such as `/var/cache/`.